30_regexp_lab/relab.txt

```diff
@@ -1,17 +1,35 @@
 Use grep to print only the lines from "input" for which:
 
-1. There is a binary number in the line
+
+1. There is a binary number in the line
 
-2. There is an even binary number in the line
+2. There is an even binary number in the line
 
-3. Line includes a hexadecimal number
+
 
-4. Line starts with a lowercase letter
+3. Line includes a hexadecimal number
 
+
+
+4. Line starts with a lowercase letter
+
+
+cat input | grep '^[a-z]'
+
+
 5. Line is not empty
 
-6. Line starts with a lowercase letter and ends in a number
+
+cat input | grep -v '^$'
+
+
+6. Line starts with a lowercase letter and ends in a number
+
+
 
+cat input | grep ^[a-z]
+only works because I know the only lines with lowercase are the ones I actually need
 7. Line is longer than 50 characters
 
+
 8. Line is of an even length
```

Comment:
This is a hard one: `egrep ^(..)+$` - Dot is any character, so 2 dots are any 2 characters. Put that in parens and add a plus to mean couples. Now if we can partition the line into couples and nothing is left, it has an event length